src/basic/fileio.c

```diff
@@ -780,6 +781,7 @@ static int merge_env_file_push(
 
         if (!env_name_is_valid(key)) {
                 log_error("%s:%u: invalid variable name \"%s\", ignoring.", strna(filename), line, key);
+                free(value);
```

Comment:
The leak is gone. But it's hard to track `value`. Shouldn't we `free(value)` in `parse_env_file_internal`?